cloud_platform_azure/models/cloud_platform.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2016-2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import re
+import os
+
+from odoo import models, api
+from odoo.addons.cloud_platform.models.cloud_platform import FilestoreKind
+from odoo.addons.cloud_platform.models.cloud_platform import PlatformConfig
+
+
+AZURE_STORE_KIND = FilestoreKind("azure", "remote")
+
+
+class CloudPlatform(models.AbstractModel):
+    _inherit = "cloud.platform"
+
+    @api.model
+    def _filestore_kinds(self):
+        kinds = super(CloudPlatform, self)._filestore_kinds()
+        kinds["azure"] = AZURE_STORE_KIND
+        return kinds
+
+    @api.model
+    def _platform_kinds(self):
+        kinds = super(CloudPlatform, self)._platform_kinds()
+        kinds.append("azure")
+        return kinds
+
+    @api.model
+    def _config_by_server_env_for_exoscale(self):
+        fs_kinds = self._filestore_kinds()
+        configs = {
+            "prod": PlatformConfig(filestore=fs_kinds["azure"]),
+            "integration": PlatformConfig(filestore=fs_kinds["azure"]),
+            "labs": PlatformConfig(filestore=fs_kinds["azure"]),
+            "test": PlatformConfig(filestore=fs_kinds["db"]),
+            "dev": PlatformConfig(filestore=fs_kinds["db"]),
+        }
+        return configs
+
+    @api.model
+    def _check_filestore(self, environment_name):
+        params = self.env["ir.config_parameter"].sudo()
+        use_azure = (params.get_param("ir_attachment.location") ==
+                     AZURE_STORE_KIND.name)
+        if environment_name in ("prod", "integration"):
+            # Labs instances use azure by default, but we don't want
+            # to enforce it in case we want to test something with a different
+            # storage. At your own risks!
+            assert use_azure, (
```

Comment:
why not logger?

Reply:
Because we want to stop Odoo from starting if the check fails